src/message/message.rs

```diff
@@ -0,0 +1,270 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+
+use super::Attachment;
+
+///  Wrapper for plain message. Provides helpers for message building and packing/unpacking.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Message {
```

Comment:
`from_prior` field needs to be supported

Reply:
I plan to define at when i start from_prior implementation